### PR TITLE
Search refactor and input check

### LINKED ISSUE #109: Refactor search bar, add user input invalidation
## Expected Behavior
App renders Searched after finished typing and click search or press 'enter'.

## Actual/Current Behavior
App renders right after typing a character in searchbar.

## Testing Steps
  1. Search pokemon. Try all lowercased, uppercase, capitalized input.
  2. Check pokemon Abouts tab flavor text.
  3. Observe.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Changed conditional rendering of Searched component in PokeList.
2. Added pokemon name length check(greater or equal to 3 letters) of  search input check.
3. Gave `getSearchBarDataApi` a `name` parameter instead of using prop for it.
4. Passed `pokemonName` to Header to use as argument for search function.

## Purpose
Prevent rendering when user hasn't finished searching or user enters nothing.

## Approach
Add conditional to check search bar input length.

Closes #109

### CODE REVIEW

File: src/components/PokeApp.js [hunk 1]
```diff
@@ -53,11 +52,10 @@ function PokeApp() {
       Promise.all(pokemonPromises).then((data) => {
         // make copy previous state, add more pokemon objects, then set state with more data
          setAllPokemon([...allPokemons, ...data]);
-         setFetchedData([...allPokemons, ...data]);
+         //setFetchedData([...allPokemons, ...data]);
```
Comment (BlackCubes) [inline src/components/PokeApp.js:55]: What's dis comment doing here 👀???

**Review by BlackCubes — APPROVED** [review]
Looks really good! I noticed that the diff size for `package-lock.json` is about 21,000, and wondering if that's kewl with you? Otherwise, noice!

**Comment by Lap343** [comment]
Ok Kendrick wants us to point this stuff out. On one of Danny's last PR he removed 20K+ code and now it looks like you are adding 20K+ code back in. Is that what you want?

File: src/components/Header.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
+       // invoke api search function in pokeapp
+    setUserDidSearch(true);
+    getSearchBarDataApi(pokemonName);
+    }
+
   };
   const enterSubmit = (e) => {
-    if (e.key === "Enter") {
-      props.getSearchBarDataApi();
+    if (e.key === "Enter" && pokemonName.length>=3) {
```
Comment (krferrell) [inline src/components/Header.js:19]: Nice checking here.

**Review by krferrell — APPROVED** [review]
Great work!

**Pull request merged**